File: _CodeTopics/LeetCode_contest/biweekly/biweekly2020/28/28_1.py
class Solution(object):
    def finalPrices(self, prices):
        """
        :type prices: List[int]
        :rtype: List[int]
        """
        
        length = len(prices)
        if length == 1:
            return prices
        
        res = [None] * length
        for i in range(length):
            res[i] = prices[i]
            for j in range(i+1,length):
                if prices[j] <= prices[i]:
                    res[i] = prices[i] - prices[j]
                    break
        return res
    
"""
https://leetcode-cn.com/submissions/detail/78709545/

103 / 103 个通过测试用例
	状态：通过
执行用时：32 ms
内存消耗：12.9 MB
"""

File: _CodeTopics/LeetCode/1401-1600/001475--biweekly28_1/tran/28_1.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {

        // 第 28 场双周赛第一题
        // https://leetcode-cn.com/submissions/detail/78709545/

        int n = prices.size();
        if (n == 1)
            return prices;

        vector<int> res(n, 0);
        for (int i = 0; i < n; ++i) {
            res[i] = prices[i];
            for (int j = i+1; j < n; ++j) {
                if (prices[j] <= prices[i]) {
                    res[i] = prices[i] - prices[j];
                    break;
                }
            }
        }
        return res;
    }
};

/*
https://leetcode.cn/submissions/detail/357836686/

执行用时：
4 ms
, 在所有 C++ 提交中击败了
81.96%
的用户
内存消耗：
9.5 MB
, 在所有 C++ 提交中击败了
96.20%
的用户
通过测试用例：
103 / 103
*/
